docs/root/configuration/listeners/udp_filters/udp_proxy.rst

```diff
@@ -0,0 +1,134 @@
+.. _config_udp_listener_filters_udp_proxy:
+
+UDP proxy
+=========
+
+.. attention::
+
+  UDP proxy support should be considered alpha and not production ready.
+
+* :ref:`v2 API reference <envoy_api_msg_config.filter.udp.udp_proxy.v2alpha.UdpProxyConfig>`
+* This filter should be configured with the name *envoy.filters.udp_listener.udp_proxy*
+
+Overview
+--------
+
+The UDP proxy listener filter allows Envoy to operate as a *non-transparent* proxy between a
+UDP client and server. The lack of transparency means that the upstream server will see the
+source IP and port of the Envoy instance versus the client. All datagrams flow from the client, to
+Envoy, to the upstream server, back to Envoy, and back to the client.
+
+Because UDP is not a connection oriented protocol, Envoy must keep track of a client's *session*
+such that the response datagrams from an upstream server can be routed back to the correct client.
+Each session is index by the 4-tuple consisting of source IP/port and local IP/port that the
+datagram is received on. Sessions last until the :ref:`idle timeout
+<envoy_api_field_config.filter.udp.udp_proxy.v2alpha.UdpProxyConfig.idle_timeout>` is reached.
+
+Load balancing and unhealthy host handling
+------------------------------------------
+
+Envoy will fully utilize the configured load balancer for the configured upstream cluster when
+load balancing UDP datagrams. When a new session is created, Envoy will associate the session
+with an upstream host selected using the configured load balancer. All future datagrams that
+belong to the session will be routed to the same upstream host.
+
+When an upstream host becomes unhealthy (due to :ref:`active health checking
+<arch_overview_health_checking>`), Envoy will attempt to create a new session to a healthy host
+when the next datagram is received.
+
+Circuit breaking
+----------------
+
+The number of sessions that can be created per upstream cluster is limited by the cluster's
+:ref:`maximum connection circuit breaker <arch_overview_circuit_break_cluster_maximum_connections>`.
+By default this is 1024.
+
+Example configuration
+---------------------
+
+The following example configuration will cause Envoy to listen on UDP port 1234 and proxy to a UDP
+server listening on port 1235.
+
+  .. code-block:: yaml
+
+    admin:
+    access_log_path: /tmp/admin_access.log
+    address:
+      socket_address:
+        protocol: TCP
+        address: 127.0.0.1
+        port_value: 9901
```

Comment:
A cut and paste of the YAML doesn't work as the stanzas under `admin:` need to be indented. Possible that more YAML fixes required.

Reply:
Thanks will fix. I keep having this issue with the editor I use for docs (atom).